### ISSUE TITLE
Potential hang when `LEFT JOIN` a subquery with `EXP`

### ISSUE THREAD

**Issue by suyZhong** [crate/crate#16952]
### CrateDB version

CrateDB 5.10.0-SNAPSHOT built 87c6dfa

### CrateDB setup information

Manual build following the instructions here: https://github.com/crate/crate/blob/master/devs/docs/basics.rst

### Problem description

Consider the following test case. The query causes the server to hang. Things go well if changing `EXP(-1110102730.1852759636)` to `EXP(-1110102730)` for example.

### Steps to Reproduce

```sql
DROP TABLE IF EXISTS t0;

CREATE  TABLE  t0( c1 FLOAT ); 

-- ok
SELECT * FROM t0 LEFT JOIN (SELECT * FROM t0 WHERE ((EXP(-1110102730)) <> t0.c1)) AS sub0  ON true;

-- hang
SELECT * FROM t0 LEFT JOIN (SELECT * FROM t0 WHERE ((EXP(-1110102730.1852759636)) <> t0.c1)) AS sub0  ON true;
```

### Actual Result

As mentioned above.

### Expected Result

As mentioned above.

**Comment by BaurzhanSakhariev** [comment]
Thanks for the report @suyZhong!

There are multiple problems here:
    
1.   Integral argument:
     `SELECT EXP(-1110102730)` --> CrateDB returns 0
       Postgres returns `Query Error: value out of range: underflow`

     `SELECT EXP(1110102730) `--> CrateDb returns "Infinity".
      Postgres returns `Query Error: value out of range: overflow`

     // Floating point argument
     `SELECT EXP(-1110102730.1852759636)` // CrateDB returns 8.605977803671129013336162895837447E-482111491
    
Postgres returns `0.0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000`

I think this one is fine, it's essentially 0 with different precision.

  `SELECT EXP(1110102730.1852759636)` --> CrateDB returns 1.161983010894382095164354623117694E+482111490

   CrateDb output seems fine and we return double precision as documented 
   https://www.wolframalpha.com/input?i=EXP%281110102730.1852759636%29
 
   Postgres returns Query Error: value overflows numeric format

   https://www.db-fiddle.com/f/97ciQwsQWGV8t8dpRjWBJ4/0

Whether we want to align with PG in cases where we don't throw is a different topic.

    
2. Hang is not related to execution, it happens when applying optimizer, during analysis. I'm checking it now and will post a workaround if there is any.

**Comment by BaurzhanSakhariev** [comment]
Problematic query works (doesn't hang and returns an empty result set) in 5.8.5 and 5.9.2, looks like a regression. 
Reverting recent nested-loop-as-hash-join doesn't help, still checking.

**Comment by BaurzhanSakhariev** [comment]
Thanks for reporting @suyZhong,  fix will be available in the 5.9.3 hotfix release.
